### ISSUE TITLE
Manually iterate to next_run or next_paragraph

### ISSUE THREAD

**Issue by andrewnc** [python-openxml/python-docx#293]
So I'm working on building a pseudo-templating language for Word here at work. I've been able to use python-docx very successfully thus far. The current functionality I'm trying to implement is an {% if val %} conditional block. It would delete any paragraphs or runs between it and the nearest {% endif %} tag. This allows for one document template to generate many types of documents by including these nice conditionals in the Word template. 

So, basically, I'm looking for a way to remove all paragraphs between my {% if %} tag and my {% endif %} tag in the word document.

I was thinking of doing this with a next_run or next_paragraph tag and just using a while text != {% endif %}, but there doesn't seem to be support for this yet. 

Thanks for any help!

**Comment by scanny** [comment]
have you looked at #40?

I think you'll need to get more specific as far as where you're having trouble. This specific functionality (templating) has been done by a few folks, but there are no specific active enhancement proposals (pull requests) for it that I know of.

The templating problem is pretty hairy in the general case, because Word will break up runs at arbitrary points for spell checking, adding or correcting text, etc.

Those who have had success with this that I know of have used either whole-paragraph markers or "curated" templates that keep their markers all in a single run.
